File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/time/TimeFormattingChain.java
package com.mapbox.services.android.navigation.v5.utils.time;


class TimeFormattingChain {

  TimeFormatResolver setup(boolean isDeviceTwentyFourHourFormat) {
    TimeFormatResolver noneSpecified = new NoneSpecifiedTimeFormat(isDeviceTwentyFourHourFormat);
    TimeFormatResolver twentyFourHours = new TwentyFourHoursTimeFormat();
    twentyFourHours.nextChain(noneSpecified);
    TimeFormatResolver rootOfTheChain = new TwelveHoursTimeFormat();
    rootOfTheChain.nextChain(twentyFourHours);

    return rootOfTheChain;
  }
}

File: libnavigation-model/src/main/java/com/mapbox/navigation/model/formatter/time/TimeFormattingChain.kt
package com.mapbox.services.android.navigation.v5.utils.time

internal class TimeFormattingChain {

    fun setup(isDeviceTwentyFourHourFormat: Boolean): TimeFormatResolver {
        val noneSpecified = NoneSpecifiedTimeFormat(isDeviceTwentyFourHourFormat)
        val twentyFourHours = TwentyFourHoursTimeFormat(noneSpecified)
        return TwelveHoursTimeFormat(twentyFourHours)
    }
}
